### ISSUE TITLE
Set element indices in previews don't match input indices

### ISSUE THREAD

**Issue by VenelinMartinov** [pulumi/pulumi-terraform-bridge#2527]
When previewing changes to set elements, the indices in the state won't match these in the user inputs. This can lead to slightly confusing diffs:

```
props1:                   []string{"val2", "val3", "val1"},
props2:                   []string{"val2", "val1"},
```
produces

```
    ~ prov:index/test:Test: (update)
        [id=newid]
        [urn=urn:pulumi:test::test::prov:index/test:Test::mainRes]
      ~ tests: [
          - [2]: "val3"
        ]
```

notice that the diff is flagged against index 2 but it is actually triggered by index 1 because the right value is there.

related to https://github.com/pulumi/pulumi-terraform-bridge/issues/2281

**Comment by t0yv0** [comment]
My 2c is that it is not worth fixing the indices unless either https://github.com/pulumi/pulumi/issues/17556  or first-class set support lands in the engine. This is relatively brittle to support in the bridge. We can see how much user confusion questions we'll get on this, but noting that this diff is showing the right element being removed which is super helpful and a big improvement over how it worked before accurate bridged previews.

**Comment by VenelinMartinov** [comment]
Yeah, I think it's not worth fixing either and it is not currently possible in the general case without bigger changes in the engine.
